streams/client/client.go

```diff
@@ -30,12 +30,60 @@ type Client struct {
 	path string
 }
 
-type response struct {
-	peerCore.ID
+type Request struct {
+	client     *Client
+	to         []peerCore.ID
+	cmd        string
+	body       command.Body
+	cmdTimeout time.Duration
```

Comment:
Why not ctx instead of timeout

Reply:
coz network read/write are blocking using time